tests/integration/integration_go_test.go

```diff
@@ -39,6 +39,27 @@ func TestNoEmitExitStatus(t *testing.T) {
 	})
 }
 
+// This checks that error logs are not being emitted twice
+func TestNoLogError(t *testing.T) {
+	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
+	integration.ProgramTest(t, &integration.ProgramTestOptions{
+		Dir: filepath.Join("go", "go-exit-error"),
+		Dependencies: []string{
+			"github.com/pulumi/pulumi/sdk/v3",
+		},
+		Stdout:        stdout,
+		Stderr:        stderr,
+		Quick:         true,
+		ExpectFailure: true,
+		ExtraRuntimeValidation: func(t *testing.T, stackInfo integration.RuntimeValidationStackInfo) {
+			// ensure exit status is not emitted by the program
+			errorCount := strings.Count(stderr.String()+stdout.String(), "  error: ")
```

Comment:
the previous implementation logged directly to stdout hence why the `error:` text was not red. To catch and fail on this old behavior, I instead concatenated stdout/stderr and check it for error lines.